lib/galaxy/tool_util/provided_metadata.py

```diff
@@ -109,10 +109,9 @@ def __init__(self, meta_file, job_wrapper=None):
                 try:
                     line = stringify_dictionary_keys(json.loads(line))
                     assert 'type' in line
-                except Exception:
-                    log.exception(f"({getattr(job_wrapper, 'job_id', None)}) Got JSON data from tool, but data is improperly formatted or no \"type\" key in data")
-                    log.debug(f'Offending data was: {line}')
-                    continue
```

Comment:
I wouldn't change the behavior of legacy tools. This seems like the wrong thing to do?

Reply:
Without this change running a *legacy*-provided-metadata-style tool that generates an invalid `galaxy.json` will result in a green dataset with wrong metadata without the user noticing, while running a *default*-provided-metadata-style tool that generates an invalid `galaxy.json` will result in a red dataset.
See the following table for why I think this is better:

| Scenario  | Output dataset state | Output dataset metadata |
| ------------- | ------------- | ------------- |
| Default provided metadata style + Valid default-style JSON  | Green  | OK |
| Default provided metadata style + Invalid JSON  | Red (Unable to finish job)  | N/A |
| Legacy provided metadata style + Valid legacy-style JSON  | Green  | OK |
| Legacy provided metadata style + Invalid JSON  | Green  | Wrong |
| Legacy provided metadata style + Invalid JSON + this fix | Red (Unable to finish job)  | N/A |

Reply:
> `galaxy.json` will result in a green dataset with wrong metadata

And it is also possible that it doesn't contain any / the correct data if datasets need to be moved via get_new_dataset_meta_by_basename. I think we should prefer correctness here over seemingly successful jobs.